src/woo_publications/publications/models.py

```diff
@@ -133,6 +138,43 @@ def get_owner(self) -> ActingUser | None:
         assert isinstance(log, TimelineLogProxy)
         return log.acting_user[0]
 
+    def revoke_own_published_documents(
+        self, user: User | ApiUser, remarks: str | None = None
+    ) -> None:
```

Comment:
```suggestion
    def revoke(
        self, user: User | ApiUser, remarks: str | None = None
    ) -> None:
```
maybe having a `publication.revoke()` method better encapsulates the meaning of what's happening? you can also add `self.publicatiestatus = PublicationStatus.revoked` to really make this a well encapsulated method, even though for the admin and serializer entrypoints that doesn't make a difference since they already set that field value (so it also doesn't hurt to do it).

Reply:
I named it `revoke_own_published_documents` specifically because that is exactly what this method does. I don't see the added value of revoking the publication status itself within the function because this will already happen in the admin and api. And it makes it that we can easily implement a feature in the future to revoke all documents attached to a publication if needed.

Reply:
Okay we leave it as is then - but if we ever need to support an atomic action of revoking a publication outside of the admin/API entrypoints (e.g. via a management command or celery task), then we refactor this so that the revocation is a single atomic action captured in one unit of work.